consoleApp/src/main/java/com/infoshare/service/VolunteerRepository.java

```diff
@@ -0,0 +1,74 @@
+package com.infoshare.service;
+
+import com.infoshare.domain.TypeOfHelp;
+import com.infoshare.domain.Volunteer;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+
+public class VolunteerRepository {
+
+    private static final String DB_FILE = "Volunteer.csv";
+
+    public void printFilteredList(){
+        Scanner scan = new Scanner(System.in);
+        System.out.println("Z jakiego miasta chcesz wyszukac wolontariuszy?");
+        String inputCity = scan.nextLine();
+        System.out.println("Jakiego typu pomocy szukasz?");
+        printTypesOfHelp();
+        String inputTypeOfHelp = scan.nextLine();
+        try {
+            List <Volunteer> availableVolunteers = readFromFile();
+            List <Volunteer> filteredList;
+            filteredList = availableVolunteers.stream().
+                    filter(Volunteer::isAvailable).
+                    filter(v -> v.getLocation().toLowerCase().equals(inputCity.toLowerCase())).
+                    filter(v -> v.getTypeOfHelp().getTypeOfHelp().toLowerCase().equals(inputTypeOfHelp.toLowerCase())).
+                    collect(Collectors.toList());
+
+            if (filteredList.size() == 0){
+                System.out.println("Brak dostępnych wolontariuszy");
+            }
+            else {
+                for (Volunteer volunteer : filteredList) {
+                    System.out.println(volunteer.printDescription());
+                }
+            }
+        }
+        catch (IllegalArgumentException ex)
```

Comment:
Co wyrzuca wyjatek IllegalArgumentException?

Reply:
Przy testowaniu rzucił wyjątek, że przekazany enum nie istnieje (prawdopodobnie wpisałem wtedy bez zachowania wielkości liter). Rozwiązanie try/catch nie jest idealne, jak będzie czas to spróbuję to inaczej obsłużyć.

Reply:
Ok rozumiem, pytanie bylo ktora konkretnie linijka rzuca ten wyjatek ?

Reply:
Nie pamiętam już, ale wydaje mi się, że 30.

Reply:
Hmm no wlasnie nie widac, zadna z metod po drodze nie wyrzuca tego wyjatku :) moze cos wczesniej ci wyrzucalo ten wyjatek, po czym to usunoles i teraz juz ten catch nie jest potrzebny  ?

Reply:
Możesz mieć rację, sprawdzę

Reply:
Potwierdzam, pewnie zostało z wcześniejszej wersji kodu, gdzie taki wyjątek leciał - wrzucę zaraz poprawioną, ostateczną wersję